eZ/Publish/Core/Persistence/Cache/LocationHandler.php

```diff
@@ -186,12 +186,21 @@ public function unHide($locationId)
     public function swap($locationId1, $locationId2)
     {
         $this->logger->logCall(__METHOD__, array('location1' => $locationId1, 'location2' => $locationId2));
-        $return = $this->persistenceHandler->locationHandler()->swap($locationId1, $locationId2);
+        $locationHandler = $this->persistenceHandler->locationHandler();
+
+        $return = $locationHandler->swap($locationId1, $locationId2);
+
+        $location1 = $locationHandler->load($locationId1);
+        $location2 = $locationHandler->load($locationId2);
 
         $this->cache->clear('location', $locationId1);
         $this->cache->clear('location', $locationId2);
         $this->cache->clear('location', 'subtree');
         $this->cache->clear('content', 'locations');
+        $this->cache->clear('content', $location1->contentId);
```

Comment:
Added in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/ecdbfdec19cbb86c90799aea4ab1f251f2aa4b20.